nms/app/packages/magmalte/app/components/EventAlertChart.js

```diff
@@ -37,45 +53,175 @@ const CHART_COLORS = [
   'black',
 ];
 
-const isValid = (start, end): boolean => {
-  return start.isValid() && end.isValid() && moment.min(start, end) === start;
+type DatasetFetchProps = {
+  networkId: network_id,
+  start: moment,
+  end: moment,
+  enqueueSnackbar: (msg: string, cfg: {}) => ?(string | number),
 };
 
-export default function ({startEnd}: Props) {
-  const [start, end] = startEnd;
-  const state = {
-    title: 'Frequency of Alerts and Events',
-    legendLabels: ['Alerts', 'Events'],
-  };
-  const chartStyle: ChartStyle = {
-    data: {
-      lineTension: 0.2,
-      pointRadius: 0.1,
+async function getEventAlertDataset(props: DatasetFetchProps) {
+  const {start, end, networkId} = props;
+  const [delta, unit, format] = getStep(start, end);
+  let requestError = '';
+  const queries = [];
+  const labels = [];
+
+  let s = start.clone();
+  while (end.diff(s) >= 0) {
+    labels.push(s.format(format));
+    const e = s.clone();
+    e.add(delta, unit);
+    queries.push([s, e]);
+    s = e.clone();
+  }
+
+  const requests = queries.map(async (query, _) => {
+    try {
+      const [s, e] = query;
+      const response = await MagmaV1API.getEventsByNetworkIdAboutCount({
+        networkId: networkId,
+        start: s.toISOString(),
+        end: e.toISOString(),
+      });
+      return response;
+    } catch (error) {
+      requestError = error;
+    }
+    return null;
+  });
+
+  // get events data
+  const eventData = await Promise.all(requests)
+    .then(allResponses => {
+      return allResponses.map((r, index) => {
+        const [s] = queries[index];
+
+        if (r === null || r === undefined) {
+          return {
+            t: s.unix(),
+            y: 0,
+          };
+        }
+        return {
+          t: s.unix(),
+          y: r,
+        };
+      });
+    })
+    .catch(error => {
+      requestError = error;
+      return [];
+    });
+
+  if (requestError) {
+    props.enqueueSnackbar('Error getting event counts', {
+      variant: 'error',
+    });
+  }
+
+  const alertPromResp = await MagmaV1API.getNetworksByNetworkIdPrometheusQueryRange(
+    {
+      networkId: networkId,
+      start: start.toISOString(),
+      end: end.toISOString(),
+      step: getStepString(delta, unit),
+      query: 'sum(ALERTS)',
```

Comment:
Have you tested that this gives you the number you expect? I haven't looked too much into this metric specifically and I don't know if there's redundancy or something in it where `sum` might give too high of a number. It's probably correct though.

Reply:
Unfortunately I haven't been able to verify alerts component yet. Currently it shows 2 alerts, but no alerts could be pulled through API. Once orc8r upgrades, I will verify and change it if necessary. Thanks for going through this.